packages/hash/api/src/model/knowledge/orgMembership.model.ts

```diff
@@ -0,0 +1,170 @@
+      accountId: workspaceAccountId,
+      entityId: entity.entityId,
+      version: entity.version,
+      entityTypeModel,
+      properties,
+    });
+  }
+
+  /**
+   * Get a workspace organization entity by its entity id.
+   *
+   * @param params.entityId - the entity id of the organization
+   */
+  static async getOrgMembershipById(
+    graphApi: GraphApi,
+    params: { entityId: string },
+  ): Promise<OrgMembershipModel | null> {
+    const entity = await EntityModel.getLatest(graphApi, {
+      // assumption: `accountId` of organizations is always the workspace account id
+      accountId: workspaceAccountId,
+      entityId: params.entityId,
+    });
+
+    if (
+      entity.entityTypeModel.schema.$id !==
+      WORKSPACE_TYPES.entityType.orgMembership.schema.$id
+    ) {
+      throw new Error(
+        `Entity with id ${params.entityId} is not a workspace org membership`,
+      );
+    }
+
+    return entity ? new OrgMembershipModel(entity) : null;
+  }
+
+  /**
+   * Get the org linked to the org membership.
+   */
+  async getOrg(graphApi: GraphApi): Promise<OrgModel> {
+    const { data: outgoingOrgLinks } = await graphApi.getLinksByQuery({
+      all: [
+        {
+          eq: [{ path: ["source", "id"] }, { literal: this.entityId }],
+        },
+        {
+          eq: [
+            { path: ["type", "versionedUri"] },
+            {
+              literal: WORKSPACE_TYPES.linkType.ofOrg.schema.$id,
+            },
+          ],
+        },
+      ],
+    });
+
+    const [outgoingOrgLink] = outgoingOrgLinks;
+
+    if (!outgoingOrgLink) {
+      /**
+       * This should never be the case, as the `org` link is required on `OrgMembership` entities.
+       *
+       * @todo: potentially remove this when the Graph API validates entities based on their schema
+       */
+      throw new Error(
+        `Critical: org membership with entity id ${this.entityId} doesn't have an outgoing "org" link`,
+      );
+    }
+
+    const { targetEntityId: orgEntityId } = outgoingOrgLink;
+
+    const orgModel = await OrgModel.getOrgById(graphApi, {
+      entityId: orgEntityId,
+    });
+
+    if (!orgModel) {
+      throw new Error(
+        `Critical: org membership with entity id ${this.entityId} links to non-existent org with id ${orgEntityId}`,
+      );
+    }
+
+    return orgModel;
+  }
+
+  /**
+   * Get the user linked to the org membership.
+   */
+  async getUser(graphApi: GraphApi) {
+    const { data: incomingOrgMembershipLinks } = await graphApi.getLinksByQuery(
+      {
+        all: [
+          {
+            eq: [{ path: ["target", "id"] }, { literal: this.entityId }],
+          },
+          {
+            eq: [
+              { path: ["type", "versionedUri"] },
+              {
+                literal: WORKSPACE_TYPES.linkType.hasMembership.schema.$id,
+              },
+            ],
+          },
+        ],
+      },
+    );
+
+    const [incomingOrgMembershipLink] = incomingOrgMembershipLinks;
```

Comment:
What happens here if `incomingOrgMembershipLinks` has more than one element? (which shouldn't happen unless the query is wrong)

Reply:
Because it is a required link, it should never happen - this is something relying on the Graph API to ensure.

Reply:
If the Graph API does mess up though, what happens. Will it assign `incomingOrgMembershipLink` to the first element? Or will something break, or

Reply:
Nothing is assigned here. What would happen is whichever incoming link is returned first in the array would be the user that's returned in this method.

Reply:
🤔 
`const [incomingOrgMembershipLink] = incomingOrgMembershipLinks;`

Is literally an assignment call for `incomingOrgMembershipLink`, it's destructuring `incomingOrgMembershipLinks` no?

Reply:
I think it's more robust to call `incomingOrgMembershipLinks.pop()` and maybe assert, that the list does not contain any more elements. Also, it's easier to read.

Reply:
This is what I was asking. It just assigns to the first element 
<img width="718" alt="image" src="https://user-images.githubusercontent.com/25749103/190635394-ba718ed7-8ab9-438e-9617-bfcdcce6e913.png">

Reply:
Apologies for the confusion, in my mental model I was thinking of "assigning" and "destructuring" as different things. The answer is yes, the first element is assigned to `incomingOrgMembershipLink`

Reply:
> I think it's more robust to call `incomingOrgMembershipLinks.pop()` [...] Also, it's easier to read.

@TimDiekmann at least in my experience, I've found that array destructuring is actually more common than using methods like `.pop()` in JS. But that's subjective, so if you feel strongly about it I'm happy to change it.

> maybe assert, that the list does not contain any more elements.

Can do this, but then the workspace API is writing checks for things that should be enforced by the Graph API and the schema constraints. Is this something we really want to be doing in the workspace API @Alfred-Mountfield @TimDiekmann ? And if so, which schema constraints are we checking for in the workspace API, and which are we relying on the Graph API to enforce? I think we would need some kind of boundary, but it will be difficult to define.

My instinct is to rely on the Graph API for these checks.

Reply:
My instinct is to rely on the Graph API too. I think part of the concern is we know that structural queries aren't quite doing what they quite look like they're doing at the moment. They don't quite combine constraints but more find things that match any of the constraints (I think)

But I will say that the philosophy is a little different to other places you've put checks in this PR. In some places you're checking for things and throwing errors that _shouldn't_ ever happen I believe? I'd need to dig through and look at them again, maybe they're only checking invariants created by the abstractions created in the workspace rather than invariants promised by the Graph.

Reply:
> In some places you're checking for things and throwing errors that shouldn't ever happen I believe? 

You're right that this is happening in other places of the PR, but I've only done this when I was forced to by a type error. Changes [like this one](https://github.com/hashintel/hash/commit/9a8d0016dbcc668e70191295c0088f8730e6b201) will help us remove these over time/encapsulate them in single methods.

Reply:
I'm going to resolve this because I think it's fine. @TimDiekmann feel free to re-open if you think there are outstanding issues